storage/innobase/os/os0file.cc

```diff
@@ -3216,24 +3206,26 @@ bool os_file_create_directory(const char *pathname, bool fail_if_exists) {
 for each entry.
 @param[in]	path		directory name as null-terminated string
 @param[in]	scan_cbk	use callback to be called for each entry
+@param[in]	handle_nodir	handle ENOENT error for the directory
 @param[in]	is_drop		attempt to drop the directory after scan
 @return true if call succeeds, false on error */
 bool os_file_scan_directory(const char *path, os_dir_cbk_t scan_cbk,
-                            bool is_drop) {
+                            bool handle_nodir, bool is_drop) {
   DIR *directory;
   dirent *entry;
 
   directory = opendir(path);
 
   if (directory == nullptr) {
-    os_file_handle_error_no_exit(path, "opendir", false);
+    if (errno != ENOENT || handle_nodir)
+      os_file_handle_error_no_exit(path, "opendir", false);
     return (false);
   }
 
   entry = readdir(directory);
 
   while (entry != nullptr) {
-    scan_cbk(path, entry->d_name);
+    if (!scan_cbk(path, entry->d_name)) break;
```

Comment:
Do we need the callback to determine whether to break out early? It seems even with slowrm, if one entry is not found, then the loop can immediately moved on to the next entry. With this change, the patch uses the next iteration of slowrm thread to pick up where it left off.

Is there much value to stopping early and resuming? I'm thinking of minimizing the API changes involved around the callback function, since this generates conflicts for rebases, and I don't see much value for breaking out early here.